### ISSUE TITLE
CS4014 compiler warning does not work when using null-conditional call operator

### ISSUE THREAD

**Issue by borland** [dotnet/roslyn#68911]
### Description

The [CS4014 compiler warning](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/cs4014?f1url=%3FappId%3Droslyn%26k%3Dk(CS4014)) is raised when you call a task-returning method in an async context, but forget to await it.

We found by accident that if the `?.` operator is used to call the method, the compiler warning does not flag this.

### Reproduction Steps

The following code should reproduce the issue

```csharp
using System;
using System.Threading.Tasks;

static class Program {

    static async Task Main(string[] args)
    {
        var worker = new Worker();
        // correct code
        await worker.DoSomething();
        await worker?.DoSomething();

        // incorrect code
        worker.DoSomething(); // Warning correctly raised here
        worker?.DoSomething(); // Expected Warning is not raised here. 

        await Console.Out.WriteLineAsync("All done");
    }
}

class Worker
{
    public async Task DoSomething()
    {
        await Task.CompletedTask;
    }
}
```

### Expected behavior

I expect the warning to be raised when calling using `?.`

### Actual behavior

Warning is not raised

### Configuration

I tested it using Visual Studio 2022 professional targeting both the .NET 6 and .NET 7 SDK's (specifically 6.0.410 and 7.0.203 at time of writing)

**Comment by xPaw** [comment]
I would also add to this that `await worker?.DoSomething();` should be a warning until the `await?` proposal is figured out.

Right now it's possible to end up writing code that awaits a null, and it's pretty subtle.

**Comment by borland** [comment]
@xPaw yeah; for our local codebase we ended up writing an analyzer to detect `await thing?.MethodCall()` and raising an error, forcing people to do this instead:

`var result = thing != null ? await thing.MethodCall() : null`

**Comment by sdelarosbil** [comment]
Hey, can I take this one?

**Comment by huoyaoyuan** [comment]
The expected behavior is unclear, since simply adding `await` won't do the right fix.
